source/extensions/filters/network/thrift_proxy/router/upstream_request.cc

```diff
@@ -154,6 +157,12 @@ UpstreamRequest::handleRegularResponse(Buffer::Instance& data,
       stats_.incResponseInvalidType(cluster, upstream_host_);
       break;
     }
+
+    if (draining) {
```

Comment:
simplify to `if (callbacks.responseMetadata()->isDraining()) {` ?

Reply:
not quite, we only want to do it on `status == ThriftFilters::ResponseStatus::Complete`.

Reply:
this if clause is inside `if (status == ThriftFilters::ResponseStatus::Complete) {` which ends of 167 after `onResponseComplete();`